crates/red_knot_python_semantic/resources/mdtest/terminal_statements.md

```diff
@@ -0,0 +1,640 @@
+
+def break_in_nested_then_branch(cond1: bool, cond2: bool, i: int):
+    x = "before"
+    for _ in range(i):
+        if cond1:
+            x = "loop1"
+            reveal_type(x)  # revealed: Literal["loop1"]
+        else:
+            if cond2:
+                x = "break"
+                reveal_type(x)  # revealed: Literal["break"]
+                break
+            else:
+                x = "loop2"
+                reveal_type(x)  # revealed: Literal["loop2"]
+            reveal_type(x)  # revealed: Literal["loop2"]
+        reveal_type(x)  # revealed: Literal["loop1", "loop2"]
+    reveal_type(x)  # revealed: Literal["before", "loop1", "break", "loop2"]
+
+def break_in_nested_else_branch(cond1: bool, cond2: bool, i: int):
+    x = "before"
+    for _ in range(i):
+        if cond1:
+            x = "loop1"
+            reveal_type(x)  # revealed: Literal["loop1"]
+        else:
+            if cond2:
+                x = "loop2"
+                reveal_type(x)  # revealed: Literal["loop2"]
+            else:
+                x = "break"
+                reveal_type(x)  # revealed: Literal["break"]
+                break
+            reveal_type(x)  # revealed: Literal["loop2"]
+        reveal_type(x)  # revealed: Literal["loop1", "loop2"]
+    reveal_type(x)  # revealed: Literal["before", "loop1", "loop2", "break"]
+
+def break_in_both_nested_branches(cond1: bool, cond2: bool, i: int):
+    x = "before"
+    for _ in range(i):
+        if cond1:
+            x = "loop"
+            reveal_type(x)  # revealed: Literal["loop"]
+        else:
+            if cond2:
+                x = "break1"
+                reveal_type(x)  # revealed: Literal["break1"]
+                break
+            else:
+                x = "break2"
+                reveal_type(x)  # revealed: Literal["break2"]
+                break
+        reveal_type(x)  # revealed: Literal["loop"]
+    reveal_type(x)  # revealed: Literal["before", "loop", "break1", "break2"]
+```
+
+## `raise`
+
+A `raise` statement is terminal. If it occurs in a lexically containing `try` statement, it will
+jump to one of the `except` clauses (if it matches the value being raised), or to the `else` clause
+(if none match). Currently, we assume definitions from before the `raise` are visible in all
+`except` and `else` clauses. (In the future, we might analyze the `except` clauses to see which ones
+match the value being raised, and limit visibility to those clauses.) Definitions from before the
+`raise` are not visible in any `else` clause, but are visible in `except` clauses or after the
+containing `try` statement (since control flow may have passed through an `except`).
+
+Currently we assume that an exception could be raised anywhere within a `try` block; the TODOs below
+reflect cases where we could implement a more precise understanding of where exceptions (barring
+`KeyboardInterrupt` and `MemoryError`) can and cannot actually be raised.
```

Comment:
Rewording this slightly since there are no longer TODOs below about this!

```suggestion
Currently we assume that an exception could be raised anywhere within a `try` block. We may want to
implement a more precise understanding of where exceptions (barring `KeyboardInterrupt` and
`MemoryError`) can and cannot actually be raised.
```